### PR TITLE
Parse duration according to the spec

### PR DESCRIPTION
**Pull request by ephraimkunz**
Hi, 
I'm working on a Gedcomx parsing crate (https://github.com/ephraimkunz/gedcomx-rs) and was planning on using this crate for the Gedcomx date implementation. 

While reading the date spec, I came across some examples of the "duration" format (https://github.com/FamilySearch/gedcomx/blob/master/specifications/date-format-specification.md#533-examples).

It appears this crate doesn't correctly parse durations. The current parsing code tries to parse a string like this:
`PY1M2D3TH4M5S6`

However, according to my reading of the spec that's not a valid duration string. Instead it should be:
`P1Y2M3DT4H5M6S`.

In other words, the length of the duration component should come before the specifier for which component it is, not after.

### CODE REVIEW

**Review by nicompte — APPROVED** [review]

**Comment by nicompte** [comment]
Thanks, indeed I must have read it wrong somehow :)

Good luck with your project!

I'll try to update to a more recent version of nom in the coming days.
